Fix resize on another screen, closes #14

somenewacc opened the pull request:
It only fixes the resize problem, but actually I see some other issue. (Shown on video)

somenewacc commented:
I can't record video.
Will try show on screenshots.

First situation -- 2 screens, same resolution (1920x1080). Scale on first is 100%, on second is 150%.
Can't press minimize, maximize, close buttons:
![image](https://user-images.githubusercontent.com/62895232/115427111-d3fe5d00-a209-11eb-9d24-925f4e73da10.png)
Buttons aren't clickable.

First situation -- 2 screens, same resolution (1920x1080). Scale on first is 100%, on second is 125%.
Buttons are clickable, but I can't drag window with titlebar anymore:
![image](https://user-images.githubusercontent.com/62895232/115427350-132cae00-a20a-11eb-8f37-352f3a47e551.png)

somenewacc commented:
I see some issues now:

1.
When I move window to second screen with 150% or 175% scale, I can't drag with titlebar on second screen.
But everything works with 100% and 125% scale.

2.
When window is splitted by 2 screens part on screen with 150% works fine but part on screen with 100% resizeable on all frame (will take a look if it can be fixed):
![image](https://user-images.githubusercontent.com/62895232/115444513-53e1f280-a21d-11eb-802a-43a58c7049b1.png)
![image](https://user-images.githubusercontent.com/62895232/115444525-580e1000-a21d-11eb-9d68-4c1634d46942.png)

3.
Known bug after move window to different screen:
![image](https://user-images.githubusercontent.com/62895232/115444663-868beb00-a21d-11eb-88ab-c8b7cf5bbb5b.png)
Any fixes/workarounds?

somenewacc commented:
I added commit to fix issue 3 in my list.
This commit also probably closes #10 and #7. 
c723c1c

EnterGin commented:
I rewrite code using map from global in HitTest, it works really fine now!
But there are another problems:
1. Part of window on another screen while maximized:
![image](https://user-images.githubusercontent.com/51099894/123560036-93dace00-d7a8-11eb-9abf-4c0a10a19ff0.png)

2. If I minimize window on second screen from maximized state, graphics hang after I try to show it:
![image](https://user-images.githubusercontent.com/51099894/123560089-e916df80-d7a8-11eb-8328-b47d33646f2d.png)

somenewacc commented:
I rewrite code using map from global in HitTest, it works really fine now!
But there are another problems:
1. Part of window on another screen while maximized:
![image](https://user-images.githubusercontent.com/51099894/123560036-93dace00-d7a8-11eb-9abf-4c0a10a19ff0.png)

2. If I minimize window on second screen from maximized state, graphics hang after I try to show it:
![image](https://user-images.githubusercontent.com/51099894/123560089-e916df80-d7a8-11eb-8328-b47d33646f2d.png)

somenewacc commented:
With latest commit I fixed all issues that I know so far. Now it's pretty usable (atleast for my applications).
I made it a while ago but decided to commit changes now. Some changes I don't really remember now (such as g with scale or ```WM_MOVE``` case in ```nativeEventFilter```) but it's works fine for me on any setup I tested.
Feel free to contact me if you see any issues.

UPD:
Seems like ```g.set*(g.*() * d->scaleFactor)``` makes window size incorrect on screen with scale. Removed it.
